spec/lib/index.spec.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable max-nested-callbacks */
+
+import path from "path"
+
+import _ from "lodash";
+
+import Interlock from "../../lib/index.js";
+
+const minimalValidConfig = {
+  emit: ["./index.js"],
+  root: __dirname + "/../..",
+};
+
+describe("lib/index.js", () => {
+  describe("constructor", function () {
+    // TODO: Test for [] and undefined. _.merge ignores those values.
```

Comment:
Yes, I've found `_.merge` to be unreliable.  I don't even like `_.defaults` anymore, but I'm never 100% what the output will be without consulting the documentation.  I've taken to using flatter objects and `Object.assign` lately.